src/Parser.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace CSVParser;
+
+class Parser {
+
+    /**
+     * @var path the file path to the csv file to be parsed.
+     */
+    private $path;
+
+    /**
+     * @var headers The column headers for the CSV file that give context to the column values.
+     */
+    private $headers;
+
+    /**
+     * @var delimiter the delimiter value that separates column values for this CSV file.
+     */
+    private $delimiter;
+
+    function __construct ($path, $delimiter = ',', $headers = []) 
+    {
+        /*
+         * Check that the path does exist or throws an exception
+         * Checks that the delimiter is a , | or /t
+         * Fills headers, if it's empty grabs first line from CSV instead
+         */
+
+        if(!file_exists($path)) 
+            throw new \Exception("{$path} is not a valid file.");
+        $this->path = $path; //fill path
+
+        if(!in_array($delimiter, [',', '|', '\t']))  //verify delimiter
+            throw new Exception("A delimiter of {$delimiter} is not supported.");    
+        $this->delimiter = $delimiter; //fill delimiter
+
+        $this->headers = $headers;
+        if(empty($headers)) 
+        {
+            /**
+             * @var path_open will open up path for reading
+             */
+            $path_open = fopen($path, 'r');
+            /**
+             * @var data is an array of data from the csv file
+             */
+            $data = fgetcsv($path_open, 1000, $this->delimiter);

```

Comment:
Since we cannot guarantee the max length will be 1000 characters I would set this to 0 which per the documentation means the max line length is not limited.